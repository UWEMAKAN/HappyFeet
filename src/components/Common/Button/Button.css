
.Button::before,
.Button::after,
.Button {
  width: 10rem;
  height: 3rem;
  font-size: 1.4rem;
  text-align: center;
  border: 2px solid #209CEE;
  outline: none;
  background-color: #fff;
  cursor: pointer;
  color: #209CEE;
  transition: all .2s ease-out;

  justify-self: flex-end;
}

.Button:hover {
  transform: translateY(-.5px);
  box-shadow: 0 1px 5px 2px rgba(0,0,0,.1);
}

.ButtonWrapper {
  display: flex;
}
