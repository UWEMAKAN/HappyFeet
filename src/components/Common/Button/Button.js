import React from 'react';
import propTypes from 'prop-types';
import './Button.css';

const Button = ({ styles, text, clickHandler }) => (
  <div style={styles} className="ButtonWrapper">
    <button onClick={clickHandler} className="Button" type="button">
      {text}
    </button>
  </div>
);

Button.propTypes = {
  clickHandler: propTypes.func.isRequired,
  styles: propTypes.instanceOf(Object).isRequired,
  text: propTypes.string.isRequired
};

export default Button;
