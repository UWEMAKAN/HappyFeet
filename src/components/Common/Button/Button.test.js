import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

const renderButton = (args) => {
  const defaultProps = {
    clickHandler: jest.fn(),
    styles: {},
    text: '',
    name: '',
    link: ''
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<Button {...props} />);
};

describe('Button Component', () => {
  test('renders button component', () => {
    const text = 'Button';
    const name = '1';
    const link = '/';
    const clickHandler = jest.fn();
    const wrapper = renderButton({
      text, clickHandler, name, link
    });

    wrapper.find('button').simulate('click');

    expect.assertions(3);
    expect(wrapper.find('button').text()).toEqual('Button');
    expect(clickHandler).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });
});
