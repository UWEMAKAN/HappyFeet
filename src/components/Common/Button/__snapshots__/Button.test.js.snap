// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders button component 1`] = `ShallowWrapper {}`;
